harvard-lts/drs-translation-service
# Fix duplicated consumers and connectivity issues

**Fix duplicated consumers and connectivity issues**
* * *

# What does this Pull Request do?
After doing end-to-end testing, we found some issues related to the service consumers. Sometimes more than one queue consumer was generated and this caused problems when listening to messages, in particular we found some messages that were not consumed and remained enqueued. This caused the process for some ingests to terminate at this point.

![duplicated_consumers_1](https://user-images.githubusercontent.com/17010447/175098764-8c443327-2b57-4ebd-bb23-8b7b7c1e2548.png)

The implementation of the service listener has been modified so that it is initialized at the same time as the Flask app. The new implementation is based on the listener model developed in DIMS, which has disconnection management and connection retries within the base listener itself.

The listener model is based on OOP and has been adapted to the existing code, which is more function-oriented. I have tried to impact as little as possible on the original design, focusing the changes only on what is necessary to solve the issues.

Service logging has also evolved, activating log file rotation, following the same approach as DIMS or HDHS.

# How should this be tested?

A description of what steps someone could take to:
* Execute tests through docker-compose-local.yml
* Send messages to the dims-data-ready queue and verify that only one consumer is active and that the messages are dequeued and the process flow works as expected.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- integration tests Yes

# Interested parties
@jessjass
